src/components/PostDetails.tsx

```diff
@@ -1,107 +1,154 @@
-import React from 'react';
+/* eslint-disable react-hooks/exhaustive-deps */
+import React, { useEffect, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Post } from '../types/Post';
+import { Comment } from '../types/Comment';
+import { deleteComment, getPostComments } from '../api/ClientFunctions';
 
-export const PostDetails: React.FC = () => {
-  return (
-    <div className="content" data-cy="PostDetails">
-      <div className="content" data-cy="PostDetails">
-        <div className="block">
-          <h2 data-cy="PostTitle">
-            #18: voluptate et itaque vero tempora molestiae
-          </h2>
-
-          <p data-cy="PostBody">
-            eveniet quo quis laborum totam consequatur non dolor ut et est
-            repudiandae est voluptatem vel debitis et magnam
-          </p>
-        </div>
+interface Props {
+  openedPost: Post;
+  doesFormExist: boolean;
+  setDoesFormExist: (value: boolean) => void;
+}
+
+export const PostDetails: React.FC<Props> = React.memo(
+  ({ openedPost, doesFormExist, setDoesFormExist }) => {
+    // #region states
+
+    const [comments, setComments] = useState<Comment[]>([]);
+    const [commentsError, setCommentsError] = useState(false);
+    const [areCommLoading, setAreCommLoading] = useState(false);
+
+    // #endregion
+    // #region variables
+
+    const { id, title, body } = openedPost;
+    const conditionForButton =
+      !commentsError && !areCommLoading && !doesFormExist;
+
+    // #endregion
+    // #region useEffects
+
+    useEffect(() => {
+      setAreCommLoading(true);
+
+      getPostComments(id)
+        .then(_comments => {
+          setComments(_comments);
+
+          if (commentsError) {
+            setCommentsError(false);
+          }
+        })
+        .catch(() => setCommentsError(true))
+        .finally(() => setAreCommLoading(false));
+    }, [id]);
+
+    // #endregion
+    // #region handlers
+
+    const deleteHandler = (_id: number) => {
+      const updatedComments = comments.filter(comment => comment.id !== _id);
+
+      setComments(updatedComments);
+
+      deleteComment(_id);
```

Comment:
The `deleteHandler` function updates the local state before confirming the deletion from the server. To ensure data integrity, consider updating the state only after the server confirms the deletion.